src/components/EventForm.jsx

```diff
@@ -0,0 +1,123 @@
+import Radium, { Style } from 'radium';
+import React from 'react';
+
+export default class EventForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      name: '',
+      event_start: '',
+      event_end: '',
+      recurs: false,
+    };
+
+    this.handleInputChange = this.handleInputChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleInputChange(event) {
+    const target = event.target;
+    const value = target.type === 'checkbox' ? target.checked : target.value;
+    const name = target.name;
```

Comment:
As a whole, along with the markup stuff on the bottom, logic like this could/should be abstracted into their own components so that this "parent" form component just has the single responsibility of maintaining the state of the incoming "Event" while supplying more agnostic, stateless "child" components the ability to trigger changes to its states.

Reply:
Could you elaborate here? Not sure what you mean by:

> single responsibility of maintaining the state of the incoming "Event" while supplying more agnostic, stateless "child" components the ability to trigger changes to its states.